synthtool/languages/node.py

```diff
@@ -188,9 +208,29 @@ def compile_protos(hide_output=False):
     shell.run(["npx", "compileProtos", "src"], hide_output=hide_output)
 
 
+def compile_protos_hermetic(hide_output=False):
+    """
+    Compiles protos into .json, .js, and .d.ts files using
+    compileProtos script from google-gax.
+    """
+    logger.debug("Compiling protos...")
+    shell.run(
+        [f"{_TOOLS_DIRECTORY}/node_modules/.bin/compileProtos", "src"],
```

Comment:
@alexander-fenster as a TODO, `compileProtos` is not actually hermetic, because it performs an install step.

It would be good if we could get rid of this install step, so that we're running exactly the version of `compileProtos` installed when the container is built.